### PR TITLE
[battery_plus] Update battery_plus to 4.0.1

### PR DESCRIPTION
**Pull request by JSUYA**
* Update battery_plus 2.1.3 to 4.0.1.
* Update battery_plus_platform_interface to 1.2.2
* Update flutter_lints to 2.0.1.
* Update the example app and integration_test.

### CODE REVIEW

File: packages/battery_plus/example/pubspec.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ dev_dependencies:
     sdk: flutter
   integration_test_tizen:
     path: ../../integration_test/
-  flutter_lints: ^1.0.4
+  flutter_lints: ^2.0.1
```
Comment (swift-kim) [inline packages/battery_plus/example/pubspec.yaml:25]: This line could be removed. Please update the dependency in `battery_plus/pubspec.yaml` instead.

Comment (JSUYA) [reply packages/battery_plus/example/pubspec.yaml]: I updated code. thanks

**Review by swift-kim — APPROVED** [review]

**Comment by JSUYA** [comment]
Rebased

File: packages/battery_plus/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ## 1.1.2
 
+* Update battery_plus to 4.0.1.
```
Comment (swift-kim) [inline packages/battery_plus/CHANGELOG.md:3]: This is wrong. battery_plus_tizen 1.1.2 had already been released before this PR was merged.

**Pull request merged**